src/common/structures/apiTree.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Name } from "./name";
+import fs from "fs-extra";
+import path from "path";
+import { ramlToolLogger } from "../logger";
+
+/*
+ * Api Tree is an abstract class that represents the directory structure of several APIs
+ *
+ */
+export abstract class ApiTree {
+  metadata: { [key: string]: unknown };
```

Comment:
Don't need to do this for a class.  It will just be an empty object.  I don't want it to be `object | undefined` just an object that can be undefined.

Reply:
> Don't need to do this for a class. It will just be an empty object.

What do you mean? If you do `new ApiTree(name, "invalid path")`, you get `{ name: name, children: [] }`, without any metadata property.

> I don't want it to be `object | undefined` just an object that can be undefined.

If something can be an object or it can be undefined, then its type is `object | undefined`. It can't just be `object`, or else it will cause errors.